Client/mods/deathmatch/logic/luadefs/CLuaPlayerDefs.cpp

```diff
@@ -644,3 +650,402 @@ bool CLuaPlayerDefs::IsPlayerCrosshairVisible()
+                return false;
+
+            if (!std::holds_alternative<float>(value))
+                return false;
+
+            hud->SetComponentColor(component, static_cast<std::uint32_t>(std::get<float>(value)), property == eHudComponentProperty::FILL_COLOR_SECOND);
+            return true;
+        }
+        case eHudComponentProperty::DRAW_BLACK_BORDER:
+        {
+            if (!hud->IsComponentBar(component))
+                return false;
+
+            if (!std::holds_alternative<bool>(value))
+                return false;
+
+            hud->SetComponentDrawBlackBorder(component, std::get<bool>(value));
+            return true;
+        }
+        case eHudComponentProperty::DRAW_PERCENTAGE:
+        {
+            if (!hud->IsComponentBar(component))
+                return false;
+
+            if (!std::holds_alternative<bool>(value))
+                return false;
+
+            hud->SetComponentDrawPercentage(component, std::get<bool>(value));
+            return true;
+        }
+        case eHudComponentProperty::BLINKING_HP_VALUE:
+        {
+            if (component != HUD_HEALTH)
+                return false;
+
+            if (!std::holds_alternative<float>(value))
+                return false;
+
+            hud->SetHealthBarBlinkingValue(std::get<float>(value));
+            return true;
+        }
+        case eHudComponentProperty::DROP_COLOR:
+        {
+            if (!hud->IsComponentText(component))
+                return false;
+
+            if (!std::holds_alternative<float>(value))
+                return false;
+
+            hud->SetComponentFontDropColor(component, static_cast<std::uint32_t>(std::get<float>(value)));
+            return true;
+        }
+        case eHudComponentProperty::TEXT_OUTLINE:
+        {
+            if (!hud->IsComponentText(component))
+                return false;
+
+            if (!std::holds_alternative<float>(value))
+                return false;
+
+            hud->SetComponentFontOutline(component, std::get<float>(value));
+            return true;
+        }
+        case eHudComponentProperty::TEXT_SHADOW:
+        {
+            if (!hud->IsComponentText(component))
+                return false;
+
+            if (!std::holds_alternative<float>(value))
+                return false;
+
+            hud->SetComponentFontShadow(component, std::get<float>(value));
+            return true;
+        }
+        case eHudComponentProperty::TEXT_STYLE:
+        {
+            if (!hud->IsComponentText(component))
+                return false;
+
+            if (!std::holds_alternative<std::string>(value))
+                return false;
+
+            eFontStyle val;
+            if (!StringToEnum(std::get<std::string>(value), val))
+                return false;
+
+            // I don't know why, but calling StringToEnum causes the "hud" pointer to become invalid, leading to a crash
```

Comment:
Did you investigate with a debugger? Manually stepping through assembly code?

Reply:
No, I decided it’s not significant in the context of this PR. I resolved the crash as you can see. I can address it later and potentially fix EnumToString, but it shouldn’t have a major impact on this PR or, for example, block its merge.

Reply:
If you already have the branch and code in front of you, please invest a few minutes and debug the issue.

Reply:
The FindValue function for some reason overwrites the value of the variable on the stack. As seen here, outResult equals ``0x0177dc13``, which is the variable val, but for some reason, ``0x0177dc14`` (the variable hud) is also overwritten, as shown by the breakpoint data. I don't know why this is happening; the ASM code doesn't indicate stack overwriting. Most likely, inlining this function is a bad idea but idk

![image](https://github.com/user-attachments/assets/5cda9bb9-a230-4361-8138-f52ce542ec7d)